src/openassetio-core/include/openassetio/managerApi/ManagerInterface.hpp

```diff
@@ -697,6 +697,178 @@ class OPENASSETIO_CORE_EXPORT ManagerInterface {
+   * to query other entities or organisational structure.
+   *
+   * @note Consult the documentation for the relevant relationship
+   * traits to determine if the order of entities in the inner lists
+   * of matching references is required to be meaningful.
+   *
+   * If any relationship definition is unknown, then an empty list
+   * must be returned for that entity, and no errors raised. The
+   * default implementation returns an empty list for all
+   * relationships.
+   *
+   * @param relationshipTraitsData The traits of the relationship to
+   * query.
+   *
+   * @param entityReferences A list of @ref entity_reference
+   * "entity references" to query the specified relationship for.
+   *
+   * @param context The calling context.
+   *
+   * @param hostSession The host session that maps to the caller, this
+   * should be used for all logging and provides access to the Host
+   * object representing the process that initiated the API session.
+   *
+   * @param successCallback Callback that must be called for each
+   * successful relationship query for an input entity reference. It
+   * should be given the corresponding index of the entity reference
+   * in @p entityReferences along with a list of entity references for
+   * entities that have the relationship specified by
+   * @p relationshipTraitsData. If there are no relations, an empty
+   * list should be passed to the callback. The callback must be
+   * called on the same thread that initiated the call to
+   * `getWithRelationship`.
+   *
+   * @param errorCallback Callback that must be called for each
+   * failed relationship query for an entity reference. It should be
+   * given the corresponding index of the entity reference in
+   * @p entityReferences along with a populated
+   * @fqref{BatchElementError} "BatchElementError" (see
+   * @fqref{BatchElementError.ErrorCode} "ErrorCodes"). The callback
+   * must be called on the same thread that initiated the call to
+   * `getWithRelationship`.
+   *
+   * @param resultTraitSet A hint as to what traits the returned
+   * entities should have. May be empty.
+   *
+   * @note Ensure that your implementation of @ref managementPolicy
+   * responds appropriately when queried with relationship trait sets
+   * and a read policy to communicate to the host whether or not you are
+   * capable of handling queries for those relationships in this method.
+   */
+  virtual void getWithRelationship(const TraitsDataPtr& relationshipTraitsData,
+                                   const EntityReferences& entityReferences,
+                                   const ContextConstPtr& context,
+                                   const HostSessionPtr& hostSession,
+                                   const RelationshipSuccessCallback& successCallback,
+                                   const BatchElementErrorCallback& errorCallback,
+                                   const trait::TraitSet& resultTraitSet);
```

Comment:
[fix] As per @foundrytom's comment and subsequent offline chat, since `resultTraitSet` is no longer optional in `ManagerInterface`, it should be moved alongside the other function-specific parameter group at the beginning.

Reply:
Rearranged as part of 4165c819